registry/handlers/referrers.go

```diff
@@ -59,13 +64,25 @@ func (h *referrersHandler) GetReferrers(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	// extract the artifactType filter.
 	var annotations map[string]string
 	var artifactTypeFilter string
 	if artifactTypeFilter = r.URL.Query().Get("artifactType"); artifactTypeFilter != "" {
 		annotations = map[string]string{
 			v1.AnnotationReferrersFiltersApplied: "artifactType",
 		}
 	}
+
+	// extract the page size info. Users define page size by using the n
+	// query parameter. The default page size is 100.
+	pageSize, nParseError := strconv.Atoi(r.URL.Query().Get("n"))
+	if nParseError != nil || pageSize < minPageSize || pageSize > maxPageSize {
+		pageSize = maxPageSize
+	}
+
+	// extract the page number info.
+	pageNumber, _ := strconv.Atoi(r.URL.Query().Get("p"))
```

Comment:
The `pageNumber` is not validated. What if it is negative?

Reply:
Negative number also works here as it's just a serial marker. But I added the validation as it makes more semantic sense.